File: API/PageAndRDL/ReportWizard/C#/ReportWizard/ReportWizardState.cs
using System.Collections.Generic;
using GrapeCity.ActiveReports.Samples.ReportWizard.MetaData;

namespace GrapeCity.ActiveReports.Samples.ReportWizard 
{
	public class ReportWizardState
	{
		public ReportWizardState()
		{
			AvailableMasterReports = new List<ReportMetaData>();
			GroupingFields = new List<FieldMetaData>();
			DisplayFields = new List<FieldMetaData>();
		}

		public readonly List<ReportMetaData> AvailableMasterReports;
		
		public ReportMetaData SelectedMasterReport;
		public readonly List<FieldMetaData> GroupingFields;
		public FieldMetaData DetailGroupingField;
		public readonly List<FieldMetaData> DisplayFields;
		public bool DisplayGroupSubtotals;
		public bool DisplayGrandTotals;
	}
}

File: API/PageAndRDL/ReportWizard/VB.NET/ReportWizard/ReportWizardState.vb
Imports System.Collections.Generic
Imports GrapeCity.ActiveReports.Samples.ReportWizard.MetaData

Public Class ReportWizardState
	Public Sub New()
		AvailableMasterReports = New List(Of ReportMetaData)()
		GroupingFields = New List(Of FieldMetaData)()
		DisplayFields = New List(Of FieldMetaData)()
	End Sub

	Public ReadOnly AvailableMasterReports As List(Of ReportMetaData)
	Public SelectedMasterReport As ReportMetaData
	Public ReadOnly GroupingFields As List(Of FieldMetaData)
	Public DetailGroupingField As FieldMetaData
	Public ReadOnly DisplayFields As List(Of FieldMetaData)
	Public DisplayGroupSubtotals As Boolean
	Public DisplayGrandTotals As Boolean

End Class
